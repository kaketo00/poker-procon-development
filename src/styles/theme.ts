import { extendTheme } from '@chakra-ui/react';
import { Zen_Kaku_Gothic_New } from 'next/font/google';

const nextFont = Zen_Kaku_Gothic_New({
  weight: ['400', '700', '900'],
  subsets: ['latin'],
});

const theme = extendTheme({
  fonts: {
    heading: nextFont.style.fontFamily,
    body: nextFont.style.fontFamily,
  },
  colors: {
    brand100: {
      100: '#0DD203', // green 100%
      200: '#005CFF', // blue 100%
      300: '#B4171E', // red 100%
      600: '#777777', // dark-gray 100%
      700: '#D4D4D4', // gray 100%
      800: '#ECECEC', // light-gray 100%
    },
    brand80: {
      100: 'rgba(13, 210, 3, 0.8)', // green 80%
      200: 'rgba(0, 92, 255, 0.8)', // blue 80%
      300: 'rgba(180, 23, 30, 0.8)', // red 80%
      400: 'rgba(255, 255, 255, 0.8)', // white 80%
      500: 'rgba(0, 0, 0, 0.8)', // black 80%
      600: 'rgba(119, 119, 119, 0.8)', // gray 80%
    },
  },
  textStyles: {
    normal: {
      xs: {
        fontSize: '0.625rem',
        fontWeight: 'normal',
        lineHeight: '0.75rem',
      },
      sm: {
        fontSize: '0.75rem',
        fontWeight: 'normal',
        lineHeight: '1rem',
      },
      md: {
        fontSize: '1rem',
        fontWeight: 'normal',
        lineHeight: '1.25rem',
      },
      lg: {
        fontSize: '1.25rem',
        fontWeight: 'normal',
        lineHeight: '1.5rem',
      },
      xl: {
        fontSize: '1.5rem',
        fontWeight: 'normal',
        lineHeight: '1.75rem',
      },
    },
    bold: {
      xs: {
        fontSize: '0.625rem',
        fontWeight: 'bold',
        lineHeight: '0.75rem',
      },
      sm: {
        fontSize: '0.75rem',
        fontWeight: 'bold',
        lineHeight: '1rem',
      },
      md: {
        fontSize: '1rem',
        fontWeight: 'bold',
        lineHeight: '1.25rem',
      },
      lg: {
        fontSize: '1.25rem',
        fontWeight: 'bold',
        lineHeight: '1.5rem',
      },
      xl: {
        fontSize: '1.5rem',
        fontWeight: 'bold',
        lineHeight: '1.75rem',
      },
    },
    strong: {
      xs: {
        fontSize: '0.6875rem',
        fontWeight: 'black',
        lineHeight: '1rem',
      },
      sm: {
        fontSize: '1.25rem',
        fontWeight: 'black',
        lineHeight: '1.5rem',
      },
      md: {
        fontSize: '1.5rem',
        fontWeight: 'black',
        lineHeight: '2rem',
      },
      lg: {
        fontSize: '1.875rem',
        fontWeight: 'black',
        lineHeight: '2.375rem',
      },
      xl: {
        fontSize: '2.25rem',
        fontWeight: 'black',
        lineHeight: '2.75rem',
      },
    },
    heading: {
      h6: {
        fontSize: '1rem',
        fontWeight: 'black',
        lineHeight: '1.25rem',
      },
      h5: {
        fontSize: '1.25rem',
        fontWeight: 'black',
        lineHeight: '1.5rem',
      },
      h4: {
        fontSize: '1.5rem',
        fontWeight: 'black',
        lineHeight: '1.875rem',
      },
      h3: {
        fontSize: '1.75rem',
        fontWeight: 'black',
        lineHeight: '2.125rem',
      },
      h2: {
        fontSize: '2.25rem',
        fontWeight: 'black',
        lineHeight: '2.75rem',
      },
      h1: {
        fontSize: '1.875rem',
        fontWeight: 'black',
        lineHeight: '2.25rem',
      },
    },
    shadows: {
      light5: {
        100: '0px 0px 5px rgba(13, 210, 3, 0.8)',
        200: '0px 0px 5px rgba(0, 92, 255, 0.8)',
        300: '0px 0px 5px rgba(180, 23, 30, 0.8)',
        400: '0px 0px 5px rgba(255, 255, 255, 0.8)',
      },
      light10: {
        100: '0px 0px 10px rgba(13, 210, 3, 0.8)',
        200: '0px 0px 10px rgba(0, 92, 255, 0.8)',
        300: '0px 0px 10px rgba(180, 23, 30, 0.8)',
        400: '0px 0px 10px rgba(255, 255, 255, 0.8)',
      },
    },
  },
  breakpoints: {
    base: '0px', // 0 - 1279px game-template(S)
    sm: '80em', // 1280px - 1439px game-template(S)
    md: '90em', // 1440px - 1919px game-template(M)
    lg: '120em', // 1920px - 2559px game-template(M)
    xl: '160em', // 2560px - âˆž game-template(L)
    '2xl': '160em',
  },
});

export default theme;
