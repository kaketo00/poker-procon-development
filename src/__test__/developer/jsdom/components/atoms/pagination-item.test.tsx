import { ChakraProvider } from '@chakra-ui/react';
import { render } from '@testing-library/react';

import type { PaginationItemProps } from '@/components/atoms/pagination-item';
import PaginationItem from '@/components/atoms/pagination-item';
import theme from '@/styles/theme';

const Component = (props: PaginationItemProps) => (
  <ChakraProvider theme={theme}>
    <PaginationItem {...props} />
  </ChakraProvider>
);
describe('PaginationItem', () => {
  test('スナップショット/first - active', () => {
    const { asFragment } = render(
      <Component variant="first" number={1} isActive={true} isCurrent={false} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('スナップショット/first - not active', () => {
    const { asFragment } = render(
      <Component
        variant="first"
        number={1}
        isActive={false}
        isCurrent={false}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('スナップショット/prev', () => {
    const { asFragment } = render(
      <Component variant="prev" number={1} isActive={false} isCurrent={false} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('スナップショット/number active', () => {
    const { asFragment } = render(
      <Component
        variant="number"
        number={1}
        isActive={true}
        isCurrent={false}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('スナップショット/number - non active', () => {
    const { asFragment } = render(
      <Component
        variant="number"
        number={1}
        isActive={false}
        isCurrent={false}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('スナップショット/number current', () => {
    const { asFragment } = render(
      <Component variant="number" number={1} isActive={true} isCurrent={true} />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('スナップショット/next', () => {
    const { asFragment } = render(
      <Component
        variant="next"
        number={100}
        isActive={false}
        isCurrent={false}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('スナップショット/last = active', () => {
    const { asFragment } = render(
      <Component
        variant="last"
        number={100}
        isActive={true}
        isCurrent={false}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });

  test('スナップショット/last - not active', () => {
    const { asFragment } = render(
      <Component
        variant="last"
        number={100}
        isActive={false}
        isCurrent={false}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
