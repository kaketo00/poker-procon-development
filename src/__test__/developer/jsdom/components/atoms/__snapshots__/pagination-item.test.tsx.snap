// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationItem スナップショット/first - active 1`] = `
<DocumentFragment>
  <button
    aria-label="first"
    class="chakra-button css-mudq2d"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="chakra-icon css-u3gylf"
      focusable="false"
      viewBox="0 0 16 16"
    >
      <path
        d="M8.354 1.646a.5.5 0 0 1 0 .708L2.707 8l5.647 5.646a.5.5 0 0 1-.708.708l-6-6a.5.5 0 0 1 0-.708l6-6a.5.5 0 0 1 .708 0z"
        fill-rule="evenodd"
      />
      <path
        d="M12.354 1.646a.5.5 0 0 1 0 .708L6.707 8l5.647 5.646a.5.5 0 0 1-.708.708l-6-6a.5.5 0 0 1 0-.708l6-6a.5.5 0 0 1 .708 0z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`PaginationItem スナップショット/first - not active 1`] = `
<DocumentFragment>
  <button
    aria-label="first"
    class="chakra-button css-mudq2d"
    disabled=""
    type="button"
  >
    <svg
      aria-hidden="true"
      class="chakra-icon css-129ongp"
      focusable="false"
      viewBox="0 0 16 16"
    >
      <path
        d="M8.354 1.646a.5.5 0 0 1 0 .708L2.707 8l5.647 5.646a.5.5 0 0 1-.708.708l-6-6a.5.5 0 0 1 0-.708l6-6a.5.5 0 0 1 .708 0z"
        fill-rule="evenodd"
      />
      <path
        d="M12.354 1.646a.5.5 0 0 1 0 .708L6.707 8l5.647 5.646a.5.5 0 0 1-.708.708l-6-6a.5.5 0 0 1 0-.708l6-6a.5.5 0 0 1 .708 0z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`PaginationItem スナップショット/last = active 1`] = `
<DocumentFragment>
  <button
    aria-label="last"
    class="chakra-button css-mudq2d"
    type="button"
  >
    <svg
      aria-hidden="true"
      class="chakra-icon css-u3gylf"
      focusable="false"
      viewBox="0 0 16 16"
    >
      <path
        d="M3.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L9.293 8 3.646 2.354a.5.5 0 0 1 0-.708z"
        fill-rule="evenodd"
      />
      <path
        d="M7.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L13.293 8 7.646 2.354a.5.5 0 0 1 0-.708z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`PaginationItem スナップショット/last - not active 1`] = `
<DocumentFragment>
  <button
    aria-label="last"
    class="chakra-button css-mudq2d"
    disabled=""
    type="button"
  >
    <svg
      aria-hidden="true"
      class="chakra-icon css-129ongp"
      focusable="false"
      viewBox="0 0 16 16"
    >
      <path
        d="M3.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L9.293 8 3.646 2.354a.5.5 0 0 1 0-.708z"
        fill-rule="evenodd"
      />
      <path
        d="M7.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L13.293 8 7.646 2.354a.5.5 0 0 1 0-.708z"
        fill-rule="evenodd"
      />
    </svg>
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`PaginationItem スナップショット/next 1`] = `
<DocumentFragment>
  <button
    aria-label="next"
    class="chakra-button css-mudq2d"
    disabled=""
    type="button"
  >
    <svg
      aria-hidden="true"
      class="chakra-icon css-onkibi"
      focusable="false"
      viewBox="0 0 16 16"
    >
      <path
        d="M4.646 1.646a.5.5 0 0 1 .708 0l6 6a.5.5 0 0 1 0 .708l-6 6a.5.5 0 0 1-.708-.708L10.293 8 4.646 2.354a.5.5 0 0 1 0-.708z"
        fill="currentColor"
      />
    </svg>
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`PaginationItem スナップショット/number - non active 1`] = `
<DocumentFragment>
  <button
    aria-label="number-1"
    class="chakra-button css-1dkcmnq"
    disabled=""
    type="button"
  >
    1
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`PaginationItem スナップショット/number active 1`] = `
<DocumentFragment>
  <button
    aria-label="number-1"
    class="chakra-button css-1dkcmnq"
    type="button"
  >
    1
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`PaginationItem スナップショット/number current 1`] = `
<DocumentFragment>
  <button
    aria-label="number-1"
    class="chakra-button css-1ol6zar"
    type="button"
  >
    1
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;

exports[`PaginationItem スナップショット/prev 1`] = `
<DocumentFragment>
  <button
    aria-label="prev"
    class="chakra-button css-mudq2d"
    disabled=""
    type="button"
  >
    <svg
      aria-hidden="true"
      class="chakra-icon css-onkibi"
      focusable="false"
      viewBox="0 0 16 16"
    >
      <path
        d="M11.354 1.646a.5.5 0 0 1 0 .708L5.707 8l5.647 5.646a.5.5 0 0 1-.708.708l-6-6a.5.5 0 0 1 0-.708l6-6a.5.5 0 0 1 .708 0z"
        fill="currentColor"
      />
    </svg>
  </button>
  <span
    hidden=""
    id="__chakra_env"
  />
</DocumentFragment>
`;
